### PR TITLE
LPS-201937 Provide listTypeDefinitionsUrl

### PR DESCRIPTION
**Pull request by seltong**
Related issue: [LPS-201937](https://liferay.atlassian.net/browse/LPS-201937)

### CODE REVIEW

**Comment by seltong** [comment]
ci:test:sf

**Review by G-Sa — APPROVED** [review]
LGTM! :tada:

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/GetObjectFieldInfoMVCResourceCommand.java [hunk 1]
```diff
@@ -72,6 +75,30 @@ protected void doServeResource(
...
+							ObjectFieldConstants.BUSINESS_TYPE_PICKLIST)) {
+
+						RequestBackedPortletURLFactory
+							requestBackedPortletURLFactory =
+								RequestBackedPortletURLFactoryUtil.create(
+									resourceRequest);
+
+						return PortletURLBuilder.create(
+							requestBackedPortletURLFactory.
+								createControlPanelRenderURL(
```
Comment (marcelabc) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/GetObjectFieldInfoMVCResourceCommand.java:91]: Why did you choose the method `createControlPanelRenderURL()`?
Can you point me the pattern that you based your work?

I see that this class has a method `requestBackedPortletURLFactory.createRenderURL()`. Do you know the difference between these methods?

Comment (seltong) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/GetObjectFieldInfoMVCResourceCommand.java:91]: Hi @marcelabc, the difference between these method is very simple. The `createControlPanelRenderURL()` shows the Portlet name and the open menu. The `createRenderURL()` doesn't shows the Portlet name and the open menu.

Using the `createControlPanelRenderURL()`:
image

Using the `createRenderURL()`:
image

Comment (marcelabc) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/GetObjectFieldInfoMVCResourceCommand.java:91]: Thank you for the explanation!

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by seltong** [comment]
I'm adding the Blocked label because this PR is a dependency of [LPS-201943](https://liferay.atlassian.net/browse/LPS-201943)

**Comment by seltong** [comment]
Hey guys. We noted that this approach is not as good as we thought. I'm sending another PR :+1: 

cc: @marcelabc @G-Sa

**Comment by seltong** [comment]
Resent: https://github.com/liferay-objects/liferay-portal/pull/2885

**Pull request closed without merging**